File: src/eko-core/prompt/components/dialogue.ts
/**
 * 对话相关的提示词组件
 */

import { ComponentConfig, PromptComponent } from './types'

/**
 * 对话能力组件
 */
export const dialogueCapabilitiesComponent: ComponentConfig = {
  id: PromptComponent.DIALOGUE_CAPABILITIES,
  name: 'Dialogue Capabilities',
  description: 'Dialogue capabilities description',
  required: false,
  template: `# Terminal Environment Capabilities
You excel at helping users with:
- File operations and text editing
- Shell command execution and scripting
- Code development and project management
- System administration and automation
- Terminal-based workflows and productivity`,
  fn: async () => {
    return dialogueCapabilitiesComponent.template!
  },
}

/**
 * 对话指导原则组件
 */
export const dialogueGuidelinesComponent: ComponentConfig = {
  id: PromptComponent.DIALOGUE_GUIDELINES,
  name: 'Dialogue Guidelines',
  description: 'Dialogue guidance principles',
  required: false,
  template: `# Dialogue Guidelines
- Provide clear and helpful responses
- Ask clarifying questions when needed
- Offer practical solutions and examples
- Maintain context throughout the conversation

## Safety & Scope Confirmation
- If the user's current working directory or requested operation scope appears overly broad or system-managed (e.g., '/', '/Users', '/Users/<name>', '/home', '/var', '/etc', '/Library', OS media/library folders like 'Music Library.musiclibrary'), do not proceed with broad file operations.
- First, ask for confirmation or a narrower subpath. For example, request a specific project folder rather than scanning the entire home directory.
- Downstream in the agent execution stage, this translates to calling a human confirmation interaction (e.g., a confirm/select step) before any wide directory listing or potentially sensitive actions.`,
  fn: async () => {
    return dialogueGuidelinesComponent.template!
  },
}

File: src-tauri/src/agent/prompt/components/dialogue.rs
use async_trait::async_trait;
use std::collections::HashMap;
use std::sync::Arc;

use crate::agent::config::PromptComponent;
use crate::agent::error::{AgentError, AgentResult};
use crate::agent::prompt::components::types::{ComponentContext, ComponentDefinition};
use crate::agent::prompt::template_engine::TemplateEngine;

pub fn definitions() -> Vec<Arc<dyn ComponentDefinition>> {
    vec![
        Arc::new(DialogueCapabilitiesComponent),
        Arc::new(DialogueGuidelinesComponent),
    ]
}

struct DialogueCapabilitiesComponent;

#[async_trait]
impl ComponentDefinition for DialogueCapabilitiesComponent {
    fn id(&self) -> PromptComponent {
        PromptComponent::DialogueCapabilities
    }

    fn name(&self) -> &str {
        "Dialogue Capabilities"
    }

    fn description(&self) -> &str {
        "Dialogue capabilities description"
    }

    fn required(&self) -> bool {
        false
    }

    fn dependencies(&self) -> &[PromptComponent] {
        &[]
    }

    fn default_template(&self) -> Option<&str> {
        Some(
            r#"# Terminal Environment Capabilities
You excel at helping users with:
- File operations and text editing
- Shell command execution and scripting
- Code development and project management
- System administration and automation
- Terminal-based workflows and productivity"#,
        )
    }

    async fn render(
        &self,
        _context: &ComponentContext,
        template_override: Option<&str>,
    ) -> AgentResult<Option<String>> {
        let template = template_override
            .or_else(|| self.default_template())
            .ok_or_else(|| AgentError::Internal("missing dialogue capabilities template".to_string()))?;

        let result = TemplateEngine::new()
            .resolve(template, &HashMap::new())
            .map_err(|e| AgentError::TemplateRender(format!("failed to render dialogue capabilities template: {}", e)))?;
        Ok(Some(result))
    }
}

struct DialogueGuidelinesComponent;

#[async_trait]
impl ComponentDefinition for DialogueGuidelinesComponent {
    fn id(&self) -> PromptComponent {
        PromptComponent::DialogueGuidelines
    }

    fn name(&self) -> &str {
        "Dialogue Guidelines"
    }

    fn description(&self) -> &str {
        "Dialogue guidance principles"
    }

    fn required(&self) -> bool {
        false
    }

    fn dependencies(&self) -> &[PromptComponent] {
        &[]
    }

    fn default_template(&self) -> Option<&str> {
        Some(
            r#"# Dialogue Guidelines
- Provide clear and helpful responses
- Ask clarifying questions when needed
- Offer practical solutions and examples
- Maintain context throughout the conversation

## Safety & Scope Confirmation
- If the user's requested scope appears overly broad or system-managed (e.g., '/', '/Users', '/home', '/var'), ask for confirmation or a narrower subpath before describing large operations."#,
        )
    }

    async fn render(
        &self,
        _context: &ComponentContext,
        template_override: Option<&str>,
    ) -> AgentResult<Option<String>> {
        let template = template_override
            .or_else(|| self.default_template())
            .ok_or_else(|| AgentError::Internal("missing dialogue guidelines template".to_string()))?;

        let result = TemplateEngine::new()
            .resolve(template, &HashMap::new())
            .map_err(|e| AgentError::TemplateRender(format!("failed to render dialogue guidelines template: {}", e)))?;
        Ok(Some(result))
    }
}
